cli/src/main/kotlin/Main.kt

```diff
@@ -0,0 +1,535 @@
+import com.here.ort.scanner.scanners.ScanCode
+import com.here.ort.utils.ARCHIVE_EXTENSIONS
+import com.here.ort.utils.PARAMETER_ORDER_HELP
+import com.here.ort.utils.PARAMETER_ORDER_LOGGING
+import com.here.ort.utils.PARAMETER_ORDER_MANDATORY
+import com.here.ort.utils.PARAMETER_ORDER_OPTIONAL
+import com.here.ort.utils.encodeOrUnknown
+import com.here.ort.utils.log
+import com.here.ort.utils.packZip
+import com.here.ort.utils.printStackTrace
+import com.here.ort.utils.safeDeleteRecursively
+import com.here.ort.utils.safeMkdirs
+import com.here.ort.utils.showStackTrace
+
+import java.io.File
+
+import kotlin.system.exitProcess
+
+const val TOOL_NAME = "ort"
+
+/**
+ * The main entry point of the application.
+ */
+object Main {
+    @Parameter(description = "Enable info logging.",
+            names = ["--info"],
+            order = PARAMETER_ORDER_LOGGING)
+    private var info = false
+
+    @Parameter(description = "Enable debug logging and keep any temporary files.",
+            names = ["--debug"],
+            order = PARAMETER_ORDER_LOGGING)
+    private var debug = false
+
+    @Parameter(description = "Print out the stacktrace for all exceptions.",
+            names = ["--stacktrace"],
+            order = PARAMETER_ORDER_LOGGING)
+    private var stacktrace = false
+
+    @Parameter(description = "Display the command line help.",
+            names = ["--help", "-h"],
+            help = true,
+            order = PARAMETER_ORDER_HELP)
+    private var help = false
+
+    @Parameters(commandNames = ["analyze"], commandDescription = "Determine dependencies of a software project.")
+    private object AnalyzerCommand : Runnable {
+        private class PackageManagerConverter : IStringConverter<PackageManagerFactory> {
+            companion object {
+                // Map upper-cased package manager class names to their instances.
+                val PACKAGE_MANAGERS = PackageManager.ALL.associateBy { it.toString().toUpperCase() }
+            }
+
+            override fun convert(name: String): PackageManagerFactory {
+                return PACKAGE_MANAGERS[name.toUpperCase()]
+                        ?: throw ParameterException("Package managers must be contained in ${PACKAGE_MANAGERS.keys}.")
+            }
+        }
+
+        @Parameter(description = "The list of package managers to activate, in order.",
+                names = ["--package-managers", "-m"],
+                converter = PackageManagerConverter::class,
+                order = PARAMETER_ORDER_OPTIONAL)
+        private var packageManagers = PackageManager.ALL
+
+        @Parameter(description = "The project directory to scan.",
+                names = ["--input-dir", "-i"],
+                required = true,
+                order = PARAMETER_ORDER_MANDATORY)
+        @Suppress("LateinitUsage")
+        private lateinit var inputDir: File
+
+        @Parameter(description = "The directory to write dependency information to.",
```

Comment:
"... to write the analyzer result file to." It's more than just dependency information. Also consistently using "analyzer result" would make it easier to understand.